modopt/opt/proximity.py

```diff
@@ -345,6 +347,77 @@ def _cost_method(self, *args, **kwargs):
         return cost_val
 
 
+class SingularValueThreshold(ProximityParent):
+    r"""Singular Value Threshold operator.
+
+    This is the proximity operator solving:
+    .. math:: arg min \tau||x||_* + 1/2||x||_F
+    """
+
+    def __init__(self, threshold, initial_rank, thresh_type='soft'):
+        self.threshold = threshold
+        self.rank = initial_rank
+        self.threshold_op = SparseThreshold(
+            linear=Identity(),
+            weights=threshold,
+            thresh_type=thresh_type
+        )
+        self._incre = 5
+
+    def _op_method(self, input_data, extra_factor=1.0):
+        """Perform singular values thresholding.
+
+        Parameters
+        ----------
+        input_data : numpy.ndarray
+            Input data array
+        extra_factor : float
+            Additional multiplication factor (default is ``1.0``)
+
+        Returns
+        -------
+        data_thresholded: ndarray
+            The data with thresholded singular values.
+        """
+        ok = False
+        matrix_data = cube2matrix(input_data)
+        s = self.rank + 1
+        while not ok:
+            U, Sigma, VT = sp.sparse.linalg.svds(matrix_data, k=s)
```

Comment:
Here I would propose you instead use `modopt.signal.svd.calculate_svd` and modify this method if required. Also to avoid naming issues I would propose something along the lines of:
- `U` -> `l_singular_vec`
- `Sigma` -> `singular_vals`
- `VT` -> `r_singular_vec`